packages/office-ui-fabric-react/src/components/SwatchColorPicker/SwatchColorPicker.tsx

```diff
@@ -0,0 +1,581 @@
+    }
+  }
+
+  /**
+   * Handle the click on a cell
+   * @param item - The cell that the click was fired against
+   */
+  private _onCellClick(item: ISwatchColorPickerItemProps) {
+    if (this.props.disabled || item.disabled) {
+      return;
+    }
+
+    let index = item.index;
+
+    // If we have a valid index and it is not already
+    // selected, select it
+    if (index >= 0 && index !== this.state.selectedIndex) {
+      if (this.props.onColorChanged) {
+        this.props.onColorChanged(item.id, item.color);
+      }
+
+      this.setState({
+        selectedIndex: index,
+        expanded: false
+      });
+    } else if (index === this.state.selectedIndex) {
+
+      // The index that got the click was already selected,
+      // clear the selection
+      this._clearColors([this.props.onColorChanged, this.props.onCellHovered, this.props.onCellFocused]);
+
+      this.setState({
+        selectedIndex: -1,
+        expanded: false
+      });
+    }
+  }
+
+  /**
+   * Handle the click on a menu item
+   * @param item - The menu item that the click was fired against
+   */
+  private _onMenuItemClick(item: ISwatchColorPickerItemProps) {
+    if (this.props.disabled || item.disabled) {
+      return;
+    }
+
+    if (this.props.onMenuItemClick) {
+      this.props.onMenuItemClick(item);
+    }
+
+    // Make sure to clear any hover/focus colors
+    this._clearColors([this.props.onCellHovered, this.props.onCellFocused]);
+
+    this.setState({
+      expanded: false
+    });
+  }
+
+  /**
+   * Clear the colors by calling the given callbacks
+   * @param callbacks - The callbacks to handle the clear operation
+   */
+  private _clearColors(callbacks: ((id?: string, color?: string) => void)[]) {
+    callbacks.forEach((callback) => {
+      if (callback) {
+        callback();
+      }
+    });
+  }
+
+  /**
+   * onClick Handler for the button
+   */
+  @autobind
+  private _onButtonClick() {
+    if (this.props.disabled) {
+      return;
+    }
+
+    if (this.state.expanded) {
+      this._onDismiss();
+    }
+
+    this.setState({
+      expanded: this.props.disabled ? false : !this.state.expanded
+    });
+  }
+
+  /**
+   * Render the core of an cell or menu item
+   * @param item - The item to render
+   * @returns {JSX.Element} - Element representing the core of the item
+   */
+  private _onRenderOption(item: ISwatchColorPickerItemProps): JSX.Element {
+
+    // Menu items just need their label text
+    if (item.type !== SwatchColorPickerItemType.Cell) {
+      return <span className={ styles.menuItem } >{ item.label }</span>;
+    }
+
+    // Build an SVG for the cell with the given shape and color properties
+    return (
+      <svg className={ css(styles.svg, this.props.cellShape, this.props.cellShape.indexOf('circle') === 0 ? styles.circle : '') } viewBox='0 0 20 20' fill={ getColorFromString(item.color).str } >
+        {
+          this.props.cellShape.indexOf('circle') === 0 ?
```

Comment:
Yeah, updated. I was getting errors ( '=== cannot be applied...') when attempting to do that earlier, but I noticed I had a typo for the 'circle' aspect of cellShape which was causing that error